perf_test/graph/KokkosGraph_color_d2.cpp

```diff
@@ -210,34 +207,29 @@ int parse_inputs(KokkosKernels::Experiment::Parameters &params, int argc, char *
         else if(0 == strcasecmp(argv[i], "--algorithm"))
         {
             ++i;
-            if(0 == strcasecmp(argv[i], "COLORING_D2_MATRIX_SQUARED"))
+            if(0 == strcasecmp(argv[i], "COLORING_D2_SERIAL"))
             {
                 params.algorithm             = 1;
```

Comment:
@srajama1 Luc had the same concern, so I just made a new params struct called D2Parameters that lives in this file and only has the fields that D2 perf test uses. It also uses the correct enum types, instead of just int for everything.

Reply:
We should consider doing that for all the perf tests, honestly. I think test_common/TestParameters just started out for SPMV and SPGEMM but eventually has a ton of fields for different algorithms

Reply:
Can you put a GitHub issue on this ?

Reply:
#662